src/GeolocationContext.tsx

```diff
@@ -255,17 +257,60 @@ export const GeolocationContextProvider: React.FC = ({children}) => {
       }
     }
     checkPermission();
-  }, [appStatus]);
+  }, [appStatus, state.status]);
 
   useEffect(() => {
     locationRef.current = state.location;
   }, [state.location]);
 
-  const getCurrentPosition = useCallback(() => locationRef.current ?? null, []);
+  const getCurrentLocation = useCallback(
```

Comment:
I don't really like [using errors for flow control](https://web.archive.org/web/20140430044213/http://c2.com/cgi-bin/wiki?DontUseExceptionsForFlowControl). In this code there is nothing really wrong if the user hasn't shared location, and I think it is better to have this represented in an object with a status and location.

Reply:
Refactored to use state instead of errors :+1: